packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -482,15 +493,31 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
     /**
      * {@inheritDoc @fluidframework/datastore-definitions#(IChannel:interface).getAttachSummary}
      */
-    public getAttachSummary(fullTree: boolean = false, trackState: boolean = false): ISummaryTreeWithStats {
-        return this.summarizeCore(this.serializer);
+    public getAttachSummary(
+        fullTree: boolean = false,
+        trackState: boolean = false,
+        telemetryContext?: ITelemetryContext,
+    ): ISummaryTreeWithStats {
+        this.incrementSummarizeInstanceCount(telemetryContext);
+
+        const result = this.summarizeCore(this.serializer, telemetryContext);
+        this.increaseSummarizeTotalBlobBytes(result.stats.totalBlobSize, telemetryContext);
+        return result;
     }
 
     /**
      * {@inheritDoc @fluidframework/datastore-definitions#(IChannel:interface).summarize}
      */
-    public async summarize(fullTree: boolean = false, trackState: boolean = false): Promise<ISummaryTreeWithStats> {
-        return this.summarizeCore(this.serializer);
+    public async summarize(
+        fullTree: boolean = false,
+        trackState: boolean = false,
+        telemetryContext?: ITelemetryContext,
+    ): Promise<ISummaryTreeWithStats> {
+        this.incrementSummarizeInstanceCount(telemetryContext);
```

Comment:
I'd say that incrementSummarizeInstanceCount() is not correct - summarizeCore() can generate multiple blobs, not just one. This data should be possible to get from result.

Also - even that's small amount of code, it would be great to share it between these two methods.